source/test/ItemTest.java

```diff
@@ -8,62 +11,71 @@ class ItemTest {
     private static Item item0;
     private static Item item1;
     @BeforeAll
-    static void init(){
-        item0 = new Item(Item.ItemType.ITEM,5,"pepsi");
-        item1 = new Item(Item.ItemType.BILL,99999,"fees");
+    static void init() {
+        item0 = new Item("Pepsi",5, Item.ItemType.ITEM);
+        item1 = new Item("Fees",99999, Item.ItemType.BILL);
     }
     @Test
     void testGetId() {
-        assertEquals(0,item0.getId());
-        assertEquals(1,item1.getId());
+        assertEquals(0, item0.getId());
+        assertEquals(1, item1.getId());
     }
-
     @Test
     void testGetName() {
-        assertEquals("pepsi",item0.getName());
-        assertEquals("fees",item1.getName());
+        assertEquals("Pepsi", item0.getName());
+        assertEquals("Fees", item1.getName());
+    }
+    @Test
+    void getType() {
+        Assertions.assertEquals(Item.ItemType.ITEM, item0.getType());
+        Assertions.assertEquals(Item.ItemType.BILL, item1.getType());
     }
-
     @Test
     void testGetPrice() {
-        assertEquals(5,item0.getPrice());
-        assertEquals(99999,item1.getPrice());
+        assertEquals(5, item0.getPrice());
+        assertEquals(99999, item1.getPrice());
         assertThrows(IllegalArgumentException.class ,()->
-                new Item(Item.ItemType.ITEM,-23,"water"));
+                new Item("Water", -23, Item.ItemType.ITEM));
     }
-
-
     @Test
-    void getDatePurchased() {
+    void testBuy() {
         User user = new User("Bruce","batman","ilovecatwoman");
         Account account = new Account(user);
         account.deposit(5000);
-        item0.purchase(account);
-        assertEquals(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS),item0.getDatePurchased().truncatedTo(ChronoUnit.SECONDS));
-    }
+        assertTrue(item0.buy(account));
+        assertFalse(item1.buy(account));
 
+    }
     @Test
-    void getAccount() {
-        User user = new User("Bruce","batman","ilovecatwoman");
+    void getDatePurchased() {
+        User user = new User("Bruce", "batman", "ilovecatwoman");
         Account account = new Account(user);
         account.deposit(5000);
-        item0.purchase(account);
-        assertEquals(account,item0.getAccount());
-    }
 
+
+        assertEquals(LocalDateTime.MIN, item0.getDatePurchased());
+        assertEquals(LocalDateTime.MIN, item1.getDatePurchased());
+
+        assertTrue(item0.buy(account));
```

Comment:
remove the unnecessary assertion

Reply:
This raised an assertion error in the next function due to it not being handled right in this one. Leaving this will make sure that it was actually bought before checking its purchase date. I think leaving it is better

Reply:
To address the issue of the item not being purchased and to ensure clarity in the testing process, the getDatePurchased function should be modified to handle the problem internally. Additionally, the testing should focus solely on the getDatePurchased functionality.